cognite/client/_api/files.py

```diff
@@ -700,20 +700,30 @@ def upload_bytes(
         return FileMetadata._load(returned_file_metadata)
 
     def retrieve_download_urls(
-        self, id: Union[int, Sequence[int]] = None, external_id: Union[str, Sequence[str]] = None
+        self,
+        id: Union[int, Sequence[int]] = None,
+        external_id: Union[str, Sequence[str]] = None,
+        extended_expiration: bool = False,
     ) -> Dict[Union[int, str], str]:
         """Get download links by id or external id
 
         Args:
-            id (Union[int, Sequence[int]]: Id or list of ids.
-            external_id (Union[str, Sequence[str]]: External id or list of external ids.
+            id (Union[int, Sequence[int]]): Id or list of ids.
+            external_id (Union[str, Sequence[str]]): External id or list of external ids.
+            extended_expiration (optional, bool): Extend exipration time of download url to 1 hour.
 
         Returns:
             Dict[Union[str, int], str]: Dictionary containing download urls.
         """
         batch_size = 100
         id_batches = [seq.as_dicts() for seq in IdentifierSequence.load(id, external_id).chunked(batch_size)]
-        tasks = [dict(url_path="/files/downloadlink", json={"items": id_batch}) for id_batch in id_batches]
+        query_params = {}
+        if extended_expiration:
+            query_params["extendedExpiration"] = True
```

Comment:
is it a query parameter or parameter in POST body? mixing them in single endpoint isn't expected I think

Reply:
qparam. Agree with not mixing in single endpoint, but this was added this way to the API a while back and there are already clients depending on it.